components/brave_federated_learning/brave_operational_profiling.cc

```diff
@@ -0,0 +1,205 @@
+      this, &BraveOperationalProfiling::OnCollectionSlotStartTimerFired);
+}
+
+void BraveOperationalProfiling::LoadParams() {
+  collection_slot_size_in_minutes_ =
+      operational_profiling::features::GetCollectionSlotSizeValue();
+  simulated_local_training_step_duration_in_minutes_ = operational_profiling::
+      features::GetSimulateLocalTrainingStepDurationValue();
+  ephemeral_id_lifetime_in_days_ =
+      operational_profiling::features::GetEphemeralIdLifetime();
+
+  operational_profiling_endpoint_ = GURL(federatedLearningUrl);
+
+  LoadPrefs();
+  MaybeResetEphemeralId();
+}
+
+void BraveOperationalProfiling::LoadPrefs() {
+  platform_ = GetPlatformIdentifier();
+  last_checked_slot_ = pref_service_->GetInteger(kLastCheckedSlotPrefName);
+  ephemeral_id_ = pref_service_->GetString(kEphemeralIdPrefName);
+  ephemeral_id_expiration_time_ =
+      pref_service_->GetTime(kEphemeralIdExpirationPrefName);
+}
+
+void BraveOperationalProfiling::SavePrefs() {
+  pref_service_->SetInteger(kLastCheckedSlotPrefName, last_checked_slot_);
+  pref_service_->SetString(kEphemeralIdPrefName, ephemeral_id_);
+  pref_service_->SetTime(kEphemeralIdExpirationPrefName,
+                         ephemeral_id_expiration_time_);
+}
+
+void BraveOperationalProfiling::OnCollectionSlotStartTimerFired() {
+  simulate_local_training_step_timer_->Reset();
+}
+
+void BraveOperationalProfiling::OnSimulateLocalTrainingStepTimerFired() {
+  SendCollectionSlot();
+}
+
+void BraveOperationalProfiling::SendCollectionSlot() {
+  current_collected_slot_ = GetCurrentCollectionSlot();
+  if (current_collected_slot_ == last_checked_slot_) {
+    return;
+  }
+
+  MaybeResetEphemeralId();
+
+  auto resource_request = std::make_unique<network::ResourceRequest>();
+  resource_request->url = operational_profiling_endpoint_;
+  resource_request->headers.SetHeader("X-Brave-FL-Operational-Profile", "?1");
+
+  resource_request->credentials_mode = network::mojom::CredentialsMode::kOmit;
+  resource_request->method = "POST";
+
+  url_loader_ = network::SimpleURLLoader::Create(
+      std::move(resource_request), GetNetworkTrafficAnnotationTag());
+  url_loader_->AttachStringForUpload(BuildPayload(), "application/base64");
+
+  url_loader_->DownloadToStringOfUnboundedSizeUntilCrashAndDie(
+      url_loader_factory_.get(),
+      base::BindOnce(&BraveOperationalProfiling::OnUploadComplete,
+                     base::Unretained(this)));
+}
+
+void BraveOperationalProfiling::OnUploadComplete(
+    std::unique_ptr<std::string> response_body) {
+  if (url_loader_->ResponseInfo()->headers->response_code() == 200) {
+    last_checked_slot_ = current_collected_slot_;
+    SavePrefs();
+  }
+}
+
+std::string BraveOperationalProfiling::BuildPayload() const {
+  base::Value root(base::Value::Type::DICTIONARY);
+
+  root.SetKey("ephemeral_id", base::Value(ephemeral_id_));
+  root.SetKey("platform", base::Value(platform_));
+  root.SetKey("collection_slot", base::Value(current_collected_slot_));
```

Comment:
Are we missing the wiki link here?

Reply:
Added 👍 (it's empty for now as we're still waiting on approval)

Reply:
see https://docs.google.com/document/d/1NK3AKfJv1mmFjU-eqMQmxP9Sss58o74n8HDrSWp9lsk/edit?usp=sharing